src/cpp/session/SessionMain.cpp

```diff
@@ -699,6 +704,12 @@ Error rInit(const rstudio::r::session::RInitInfo& rInitInfo)
    if (s_printCharsetWarning)
       rstudio::r::exec::warning("Character set is not UTF-8; please change your locale");
 
+   if (!ensureLibRSoValid())
+     LOG_ERROR_MESSAGE(s_libRSoWarning +
+                       ". Linux may have loaded a different libR.so than requested."
+                       " R_HOME: " + rsession::module_context::rHomeDir() +
+                       ", R Version: " + rsession::module_context::rVersion());
+
```

Comment:
Rather than use a static which isn't thread safe, we should log this error message from within the `ensureLibRSoValid()` method.

Reply:
Additionally, should we exit here? Or at the very list emit a similar error to the R Console so that the user can be informed? IIRC, this error message will only go to our log files, and a big problem with this behaviour is that the user is unware that they're using the wrong version of R. We should probably include a note about contacting your system administrator if the versions don't match.